app/src/main/java/currencyConverter/service/CurrencyConverter.java

```diff
@@ -1,20 +1,45 @@
 package currencyConverter.service;
 
+import java.util.concurrent.Executors;
+
 import currencyConverter.exception.CurrencyRateFetchingException;
 import currencyConverter.codes.CurrencyCode;
 
-class CurrencyConverter implements ICurrencyConverter {
+public class CurrencyConverter implements ICurrencyConverter {
+
+    private static ICurrencyConverter instance;
 
     private ICurrencyRatesProvider currencyRatesProvider;
 
-    CurrencyConverter() {
+    private CurrencyConverter() {
         this.currencyRatesProvider = new CurrencyRatesProvider();
     }
 
+    public static ICurrencyConverter getInstance() {
+        if (instance == null) {
+            instance = new CurrencyConverter();
+        }
+        return instance;
+    }
+
     @Override
     public Double convert(CurrencyCode from, CurrencyCode to, double value) throws CurrencyRateFetchingException {
         Double fromRate = this.currencyRatesProvider.getCurrencyRate(from);
         Double toRate = this.currencyRatesProvider.getCurrencyRate(to);
-        return fromRate / toRate * value ;
+        return fromRate / toRate * value;
+    }
+
+    @Override
+    public void initialize() {
+        Executors.newSingleThreadExecutor().submit(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    CurrencyConverter.this.currencyRatesProvider.initialize();
+                } catch (Exception e) {
+                    //Just Ignore It
```

Comment:
Maybe It's better to leave a TODO. I think we should handle exceptions a bit smarter.